src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -62,12 +70,37 @@ public function process(ContainerBuilder $container)
                 $tmpContainer->addObjectResource($configuration);
             }
 
+            $resetPlaceholders = false;
+            if (null !== $envTypes && method_exists(BaseNode::class, 'setPlaceholder')) {
+                $defaultEnvBag = new ParameterBag($parameters);
+                foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+                    $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+                    $types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string');
+                    $default = ($hasEnv = (false === $i && $defaultEnvBag->has("env($env)"))) ? $defaultEnvBag->get("env($env)") : null;
+                    if (null !== $default && !in_array($type = gettype($default), $types, true)) {
+                        throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
+                    }
+                    $values = array();
+                    foreach ($types as $type) {
+                        $values[$type] = $hasEnv ? $default : (isset(self::$typeFixtures[$type]) ? self::$typeFixtures[$type] : null);
+                    }
+                    foreach ($placeholders as $placeholder) {
+                        $resetPlaceholders = true;
+                        BaseNode::setPlaceholder($placeholder, $values);
+                    }
+                }
+            }
+
             foreach ($exprLangProviders as $provider) {
                 $tmpContainer->addExpressionLanguageProvider($provider);
             }
 
             $extension->load($config, $tmpContainer);
 
+            if ($resetPlaceholders) {
```

Comment:
Does this really need to be conditional?
Also wondering: can this pass be used when the config component is not installed (I didn't check)?
If yes, then shouldn't BaseNode be used conditionally?

Reply:
`$resetPlaceholders` is true if we called BaseNode::setPlaceholder previously.